resolver/resolver.go

```diff
@@ -38,19 +45,21 @@ func (e ErrNoLink) Error() string {
 type ResolveOnce func(ctx context.Context, ds ipld.NodeGetter, nd ipld.Node, names []string) (*ipld.Link, []string, error)
 
 // Resolver provides path resolution to IPFS
-// It has a pointer to a DAGService, which is uses to resolve nodes.
+// It has a pointer to a FetcherConfig, which is uses to resolve nodes.
 // TODO: now that this is more modular, try to unify this code with the
 //       the resolvers in namesys
 type Resolver struct {
-	DAG ipld.NodeGetter
+	FetchConfig fetcher.FetcherConfig
 
 	ResolveOnce ResolveOnce
 }
 
 // NewBasicResolver constructs a new basic resolver.
-func NewBasicResolver(ds ipld.DAGService) *Resolver {
+func NewBasicResolver(bs blockservice.BlockService) *Resolver {
+	fc := fetcher.NewFetcherConfig(bs)
```

Comment:
is it possible we'll want a single fetcher both for pathing and other things, and will want to pass in a fetchconfig here directly? (i guess we can make a Resolver struct if we need to do that)

Reply:
It's a little tricky. `FetcherConfig` owns the `PrototypeChooser` and go-path needs a custom `PrototypeChooser` to ensure it can path into dagpb nodes using the unixfsnode ADL. We probably don't want to push responsibility for constructing the correct chooser to the caller and we can't have go-path reconfigure a shared `FetchConfig`. 

We could potentially pass the chooser into the GetSession call in go-fetcher (which would be called with a specific chooser in `go-path/BasicResolver`, but I'd want to get a better sense of the kinds of things that might need to be shared in a FetchConfig (other than blockservice) before making that kind of refactor.